File: Logitude/AngularModules/AngularModules/Common/CustomFilters/AWBDescriptionOfGoodsCustomFilter.ts
import {AppTool} from '../../Infrastructure/Tools';

export class AWBDescriptionOfGoodsCustomFilter {
    public static GetFilteredQuery(addtionalFiltersValues: any, data: any) {
        var mykeys = JSON.parse(addtionalFiltersValues);
        if (mykeys.filter(d => d.IsCustom == true)[0]) {
            for (var i in mykeys.filter(d => d.IsCustom)) {
                var propName = mykeys[i];
                if (propName.FieldName == "AirlineCode") {
                    var myAirlineCode = propName.FieldValue;
                    if (!AppTool.IsNullOrEmpty(myAirlineCode)) {
                        data = data.filter(d => d.AirlineCode == null || d.AirlineCode == myAirlineCode);
                    }
                    else {

                        data = data.filter(d => d.AirlineCode == null);
                    }
                }
            }
        }
        else {

            data = data.filter(d => d.AirlineCode == null);
        }

        return data
    }
}

File: Logitude/Logitude.BL/CommonDataModel/CustomFilters/AWBDescriptionOfGoodsCustomFilter.cs
using Simplog.Data.CommonDataModel.EntityPOCOs;
using Simplog.Server.Infrastructure.DataContracts;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Logitude.BL.CommonDataModel.CustomFilters
{
    public class AWBDescriptionOfGoodsCustomFilter
    {
        private int tenant;
        public AWBDescriptionOfGoodsCustomFilter(int tenant)
        {
            this.Tenant = tenant;
        }

        public int Tenant
        {
            get { return tenant; }
            set { tenant = value; }
        }

        public IQueryable<AWBDescriptionOfGoods> GetFilteredQuery(QueryOperations operations, IQueryable<AWBDescriptionOfGoods> queryableData)
        {
            List<QueryFilterItem> queryFilters = operations.QueryFilterItems;
            
            foreach (QueryFilterItem item in queryFilters)
            {
                if (item.IsCustom)
                {
                    if (item.FieldName == "AirlineCode")
                    {
                        string str = item.FieldValue as string;

                        if (!string.IsNullOrEmpty(str))
                        {
                            queryableData = queryableData.Where(c => c.AirlineCode == str);
                        }
                    }
                }
            }

            return queryableData;
        }
    }
}
